### PR TITLE
Search keyboard

### PR DESCRIPTION
**Pull request by vincerubinetti**

### CODE REVIEW

**Comment by vincerubinetti** [comment]
@dongbohu this is ready for you to start reviewing if you want

File: src/components/search/index.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  const onBlur = () => setFocused(false);
+  const onChangeExpanded = (expanded) => setExpanded(expanded);
+  const getClearFunc = (func) => (clearFunc.current = func);
+
+  const onKeyDown = (event) => {
+    console.log(focused, expanded, length)
+    if (!focused || expanded || !length)
+      return;
+
+
```
Comment (dongbohu) [inline src/components/search/index.js]: Is this blank line by accident?

**Review by dongbohu — COMMENTED** [review]
Only one comment.

File: src/components/search/index.js (hunk 1 above)
Comment (vincerubinetti) [reply src/components/search/index.js]: Yes

File: src/components/search/index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        break;
+    }
+  };
+
+  useEffect(() => {
+    if (focused && !expanded && length)
+      setHighlightedIndex(0);
+    else
+      setHighlightedIndex(-1);
+  }, [expanded, focused, length]);
```
Comment (dongbohu) [inline src/components/search/index.js:65]: Is `[expanded, focused, length]` on this line the return value of the arrow function? It seems unusual.

Comment (vincerubinetti) [reply src/components/search/index.js:65]: No that's a react hooks thing. Any time one of those values changes, the `useEffect` will be run. They're essentially memoization inputs.

Comment (dongbohu) [reply src/components/search/index.js:65]: My bad. I thought it was a function definition.

**Review by dongbohu — APPROVED** [review]

**Pull request merged**